docs/standard/serialization/system-text-json/source-generation.md

```diff
@@ -136,9 +138,87 @@ You can specify metadata collection mode or serialization optimization mode for
 
   :::code language="csharp" source="snippets/source-generation/csharp/MetadataOnlyNoOptions.cs" id="All":::
 
-## Specify options for serialization optimization mode
+## Source-generation support in ASP.NET Core
+
+In Blazor apps, use overloads of <xref:System.Net.Http.Json.HttpClientJsonExtensions.GetFromJsonAsync%2A?displayProperty=nameWithType> and <xref:System.Net.Http.Json.HttpClientJsonExtensions.PostAsJsonAsync%2A?displayProperty=nameWithType> extension methods that take a source generation context or `TypeInfo<TValue>`.
+
+:::zone pivot="dotnet-8-0,dotnet-7-0"
+
+In Razor Pages, MVC, SignalR, and Web API apps, use the <xref:System.Text.Json.JsonSerializerOptions.TypeInfoResolver?displayProperty=nameWithType> property to specify the context.
+
+```csharp
+[JsonSerializable(typeof(WeatherForecast[]))]
+internal partial class MyJsonContext : JsonSerializerContext { }
+```
+
+```csharp
+var serializerOptions = new JsonSerializerOptions
+{
+    TypeInfoResolver = MyJsonContext.Default;
+};
+
+services.AddControllers().AddJsonOptions(options =>
+    options.JsonSerializerOptions = serializerOptions);
+```
+
+:::zone-end
+
+:::zone pivot="dotnet-6-0"
+
+In Razor Pages, MVC, SignalR, and Web API apps, use the <xref:System.Text.Json.JsonSerializerOptions.AddContext%2A> method of <xref:System.Text.Json.JsonSerializerOptions>, as shown in the following example:
+
+```csharp
+[JsonSerializable(typeof(WeatherForecast[]))]
+internal partial class MyJsonContext : JsonSerializerContext { }
+```
+
+```csharp
+services.AddControllers().AddJsonOptions(options =>
+    options.JsonSerializerOptions.AddContext<MyJsonContext>());
+```
+
+:::zone-end
+
+:::zone pivot="dotnet-8-0"
+
+## Disable reflection defaults
+
+Because System.Text.Json uses reflection by default, calling a basic serialization method can break Native AOT apps, which can't reflect at run time. These breaks can be challenging to diagnose since apps are often debugged using the CoreCLR runtime, where the code works. Instead, if you explicitly disable reflection-based serialization, breaks are easier to diagnose. Code that uses reflection-based serialization will cause an <xref:System.InvalidOperationException> with a descriptive message to be thrown at run time.
+
+To disable default reflection in your app, set the `JsonSerializerIsReflectionEnabledByDefault` MSBuild property to `false` in your project file:
+
+```xml
+<PropertyGroup>
+  <JsonSerializerIsReflectionEnabledByDefault>false</JsonSerializerIsReflectionEnabledByDefault>
```

Comment:
```suggestion
<PropertyGroup>
<JsonSerializerIsReflectionEnabledByDefault>false</JsonSerializerIsReflectionEnabledByDefault>
```

Reply:
Why to remove the indentation here?

Reply:
Oh, didn't mean to remove that - it rendered on my end as a new empty line. Leave this as it was...